### ISSUE TITLE
upgrading deis does nothing using helm 2.1.3 and k8s 1.5

### ISSUE THREAD

**Issue by felipejfc** [kubernetes/helm#1818]
Hi, I'm trying to upgrade deis workflow from 2.9.0 to 2.10.0 using the following command:
```
$ helm upgrade v2.7.0 deis/workflow -f values-secret.yaml --version=v2.10.0 --kube-context=stag
```

The output is:

```
Release "v2.7.0" has been upgraded. Happy Helming!
LAST DEPLOYED: Thu Jan 12 13:13:43 2017
NAMESPACE: deis
STATUS: DEPLOYED

RESOURCES:
==> v1/ConfigMap
NAME                   DATA      AGE
dockerbuilder-config   2         39d
slugbuilder-config   2         39d
slugrunner-config   1         39d

==> v1/ServiceAccount
NAME            SECRETS   AGE
deis-database   1         86d
deis-logger   1         86d
deis-builder   1         86d
deis-router   1         86d
deis-logger-fluentd   1         86d
deis-nsqd   1         86d
deis-monitor-telegraf   1         86d
deis-controller   1         86d
deis-workflow-manager   1         86d

==> v1/Service
NAME                    CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
deis-monitor-influxui   100.71.72.0   <none>        80/TCP    86d
deis-nsqd   100.65.169.175   <none>    4151/TCP,4150/TCP   86d
deis-controller   100.64.63.89   <none>    80/TCP    86d
deis-monitor-grafana   100.67.119.207   <none>    80/TCP    86d
deis-database   100.64.64.189   <none>    5432/TCP   86d
deis-workflow-manager   100.65.109.172   <none>    80/TCP    86d
deis-builder   100.66.189.152   <none>    2222/TCP   86d
deis-router   100.71.77.67   a3cbf8d615b52...   80:31486/TCP,443:32764/TCP,2222:31744/TCP,9090:31943/TCP   159d
deis-monitor-influxapi   100.70.52.146   <none>    80/TCP    86d
deis-logger   100.64.153.160   <none>    80/TCP    86d
deis-logger-redis   100.67.166.148   <none>    6379/TCP   159d

==> extensions/Deployment
NAME                    DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deis-workflow-manager   1         1         1            1           159d
deis-logger-redis   1         1         1         1         101d
deis-builder   1         1         1         1         39d
deis-monitor-grafana   1         1         1         1         101d
deis-controller   1         1         1         1         39d
deis-router   1         1         1         1         159d
deis-database   1         1         1         1         101d
deis-logger   1         1         1         1         101d
deis-nsqd   1         1         1         1         101d
deis-registry-token-refresher   1         1         1         1         22d
deis-monitor-influxdb   1         1         1         1         101d

==> extensions/DaemonSet
NAME                    DESIRED   CURRENT   READY     NODE-SELECTOR   AGE
deis-monitor-telegraf   4         4         4         <none>          86d
deis-logger-fluentd   4         4         4         <none>    86d

==> v1/Secret
NAME                    TYPE      DATA      AGE
objectstorage-keyfile   Opaque    6         86d
registry-secret   Opaque    4         22d
minio-user   Opaque    2         39d
deis-router-dhparam   Opaque    1         86d
```

deis-upgrade job's pod log is:

```
2017-01-12T15:13:53.843464151Z pods for deis-logger-fluentd daemonset deleted successfuly
2017-01-12T15:13:53.845564546Z pods for deis-registry-proxy daemonset deleted successfuly
2017-01-12T15:13:53.846767887Z pods for deis-monitor-telegraf daemonset deleted successfuly
```

But, despite it says the upgrade succeeded and the output of 
```
$ helm history v2.7.0 --kube-context=stag
```

is

```
REVISION	UPDATED                 	STATUS    	CHART
1       	Sun Dec  4 12:32:09 2016	SUPERSEDED	workflow-v2.7.0
2       	Sun Dec  4 12:35:04 2016	SUPERSEDED	workflow-v2.8.0
12      	Wed Dec 21 02:14:39 2016	SUPERSEDED	workflow-v2.9.0
24      	Thu Jan 12 13:13:43 2017	DEPLOYED  	workflow-v2.10.0
```

No pod or deployment has been updated, they are still pointing to 2.9.0 version

I'm using kubernetes 1.5.1

**Comment by felipejfc** [comment]
Ok, it seems that the problem was that I was trying to jump from version 2.9.0 directly to 2.10.0 and it was refusing to upgrade somehow, then I tried going from 2.9.0 to 2.9.1 first then 2.10.0 and it worked!
If we can't really jump versions I think it should be documented somewhere

**Comment by bacongobbler** [comment]
pretty sure it's due to a bug we fixed in our chart in v2.9.1, which seems to line up with your story. https://github.com/deis/workflow/commit/b3a9db14c41caae331d53ef5dd553d856ccbb271

I think we can close this :)
